lib/http/cookie.rb

```diff
@@ -655,6 +655,13 @@ def <=> other
   end
   include Comparable
 
+  # Hash serialization helper for use back into other libraries (Like Selenium)
+  def to_h
+    {}.tap do |hash|
+      PERSISTENT_PROPERTIES.each { |property| hash[property.to_sym] = send(property) }
+    end
```

Comment:
You can do this as a one-liner using `each_with_object`:

```
PERSISTENT_PROPERTIES.each_with_object({}) { |prop, h| h[prop] = instance_variable_get("@#{prop}") }
```

A couple of things to note:

- This keeps the keys as strings. If there's a compelling reason to make them symbols let me know!
- This follows the example of the YAML methods and retrieves an instance variable value instead of using an accessor method. Probably there's no difference, but I like to follow existing conventions in old codebases.

If this makes sense to you, I'd like to see a test for this method that demonstrates that the keys are strings, and that all the persistent properties are included in the hash. Then I think it would be ready to merge!

Reply:
Re point 2. Happy to agree on this no issues. I'll get this change made.

RE point 1. I would say it's more conventional to hash symbolized keys in a hash. Because given this is likely to go to a JSON payload and/or cookie, **[they need to be accepted there as symbols](https://github.com/SeleniumHQ/selenium/blob/trunk/rb/lib/selenium/webdriver/common/manager.rb#L45-L62)**. This reduces the friction.

The previous work I did was about making the friction seamless from fetched cookie to the cookie object/store i.e. the flow is "Selenium -> grab cookie -> Co-erce into HTTP::Cookie"

Here the work is the opposite "HTTP::CookierJar -> grab cookie -> Co-erce into a legitimate web based cookie

Hopefully that explains some of my thought processes